unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentView.cs

```diff
@@ -271,21 +319,121 @@ public void SetCollectibleEmotes(WearableItem[] wearables)
                         };
 
-                        nftIds[j] = wearables[i + j].id;
+                        nftIds[j] = (wearables[i + j].id, "emote");
+                        wearableItemsForThisPage[j] = wearables[i + j];
                     }
                     else
                     {
                         pageElements[j] = null;
-                        nftIds[j] = "";
+                        nftIds[j] = ("", "");
+                        wearableItemsForThisPage[j] = null;
                     }
                 }
                 nftPageView.SetPageElementsContent(pageElements, nftIds);
+                nftPageView.IsNftInfoActived = true;
+                nftPageView.ConfigureNFTItemInfo(nftItemInfo, wearableItemsForThisPage, false);
                 nftPageView.OnClickBuyNft += ClickOnBuyWearable;
+                nftPageView.OnFocusAnyNtf += FocusAnyNtfItem;
+                ownedNftEmotePageViews.Add(nftPageView);
                 nftEmotesCarousel.AddItem(nftPageView);
             }
             nftEmotesCarousel.GenerateDotsSelector();
         }
 
+        public void SetCollectibleNames(NamesResponse.NameEntry[] names)
+        {
+            nftNamesCarousel.gameObject.SetActive(names.Length > 0);
+            nftNamesCarousel.CleanInstantiatedItems();
+            emptyNamesText.SetActive(names.Length <= 0);
+            for (int i = 0; i < names.Length; i += 4)
+            {
+                PoolableObject nftPagePoolElement = nftPagesEntryPool.Get();
+                nftPagesPoolableQueue.Add(nftPagePoolElement);
+                nftPagePoolElement.gameObject.transform.SetParent(nftNamesCarouselContent, false);
+                NftPageView nftPageView = nftPagePoolElement.gameObject.GetComponent<NftPageView>();
+                nftPageView.OnClickBuyNft -= ClickOnBuyWearable;
+                nftPageView.OnFocusAnyNtf -= FocusAnyNtfItem;
+                NFTIconComponentModel[] pageElements = new NFTIconComponentModel[4];
+                (string, string)[] nftIds = new (string, string)[4];
+                for (int j = 0; j < 4; j++)
+                {
+                    if (names.Length > i + j && names[i + j] != null)
+                    {
+                        pageElements[j] = new()
+                        {
+                            showMarketplaceButton = true,
+                            showType = true,
+                            type = "name",
+                            marketplaceURI = "",
+                            name = names[i + j].Name,
+                            rarity = "legendary",
+                            imageURI = ""
+                        };
+
+                        nftIds[j] = (names[i + j].ContractAddress, "name");
+                    }
+                    else
+                    {
+                        pageElements[j] = null;
+                        nftIds[j] = ("", "");
+                    }
+                }
```

Comment:
hiding low-level details in the descriptive high-level function would help readability here

Reply:
done